test-data/unit/check-based-typevar.test

```diff
@@ -0,0 +1,49 @@
+[case testConstrainedTypeVarWithVariance]
+from typing import Protocol, TypeVar
+
+T1 = TypeVar("T1", int, str)
+T2 = TypeVar("T2", int, str, covariant=True)  # E: TypeVar with constraints cannot have variance  [valid-type]
+T3 = TypeVar("T3", int, str, contravariant=True)  # E: TypeVar with constraints cannot have variance  [valid-type]
+T4 = TypeVar("T4", int, str, infer_variance=True)  # E: Unexpected argument to "TypeVar()": "infer_variance"  [misc] \
+                                                   # E: "int" not callable  [operator]  # stupid fake stubs moment
+
+class A(Protocol[T1]):
+    """inferred as covariant, but it's constrained so it's valid"""
+    def f(self) -> T1:
+        pass
+
+
+[case testTypeVarWrongNameStillWorks]
+from typing import TypeVar
+
+T = TypeVar("NotT")  # E: String argument 1 "NotT" to TypeVar(...) does not match variable name "T"  [misc]
```

Comment:
is this test supposed to include a constraint and variance argument? otherwise it seems unrelated to this change

Reply:
it's just a small improvement to not break the typevar completely when the name is wrong